vxquery-core/src/main/java/org/apache/vxquery/datamodel/builders/jsonItem/ArrayBuilder.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.vxquery.datamodel.builders.jsonItem;
+
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.hyracks.data.std.api.IMutableValueStorage;
+import org.apache.hyracks.data.std.api.IValueReference;
+import org.apache.hyracks.data.std.util.ArrayBackedValueStorage;
+import org.apache.vxquery.datamodel.values.ValueTag;
+import org.apache.vxquery.util.GrowableIntArray;
+
+public class ArrayBuilder {
```

Comment:
It seems like ArrayBuilder, ObjectBuilder and SequenceBuilder could all share an abstract class. Do you think this would be useful? I guess it would only hold reset. You might be able to create a version of finish that could be shared, between array and object builder.